contracts/Collector.sol

```diff
@@ -68,17 +86,27 @@ contract Collector is ICollector{
     }
 
     modifier validShares(RevenuePartner[] memory _partners){
-        uint totalShares = 0;            
-        for(uint i = 0; i < _partners.length; i++)
+        uint totalShares = 0;
+
+        for(uint i = 0; i < _partners.length; i++){
             totalShares = totalShares + _partners[i].share;
+            require(_partners[i].share > 0, "0 is not a valid share");
+        }
```

Comment:
for loops are super expensive.. could we use different strategy?

Reply:
I think it's out of the scope of this task and we should create a new task for this.